test/util/error/error.go

```diff
@@ -1,21 +1,47 @@
 package error
 
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
 import (
+	"regexp"
 	"slices"
 	"testing"
 )
 
-// Copyright (c) Microsoft Corporation.
-// Licensed under the Apache License 2.0.
+type assertOptions int
+
+const trimWhitespace assertOptions = 1
+
+func TrimWhitespace() assertOptions {
+	return trimWhitespace
+}
 
 // AssertErrorMessage asserts that err.Error() is equal to wantMsg.
-func AssertErrorMessage(t *testing.T, err error, wantMsg string) {
+func AssertErrorMessage(t *testing.T, err error, wantMsg string, opts ...assertOptions) {
```

Comment:
Can we introduce unit tests for the function?

Reply:
It's a bit hard when it's testing.T code. I did verify that it works as I was using it to help debug the issue :)